### ISSUE TITLE
Support for zone_redundant in azurerm_eventhub_namespace in azurerm v4

### ISSUE THREAD

**Issue by tw-sematell** [hashicorp/terraform-provider-azurerm#27239]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

The zone_redundant setting for azurerm_eventhub_namespace was removed in azurerm v4. As the platform engineer, I want to be able to provision zone_redundant Kafka services in Azure without using additional tooling as a workaround.

### New or Affected Resource(s)/Data Source(s)

azurerm_eventhub_namespace

### Potential Terraform Configuration

```hcl
resource "azurerm_eventhub_namespace" "example" {
  name                = "evhns-example"
  resource_group_name = azurerm_resource_group.services.name
  location            = azurerm_resource_group.services.location
  
  zone_redundant = true
  sku            = "Standard"
  capacity       = 1
}
```

**Comment by rcskosir** [comment]
Linking the PR that removed `zone_redundant` for context: https://github.com/hashicorp/terraform-provider-azurerm/pull/26611

**Comment by deese** [comment]
Now when you create a new namespace it is created without the zone redundancy enabled but if you create the namespace from the GUI with the same config, it is enabled based on the location. 
How should it be done now then? 

In the link PR mentions Premium eventhub but in Microsoft documentation states that also Standard has the zone redundancy enabled (and in the GUI works, as mentioned)

**Comment by WodansSon** [comment]
@nerddtvg, that is exactly right, this field is actually deprecated and is being change for all accounts directly in Azure by the service team. There was some logic added to the Service, to not break Terraform and cause a diff in 3.x providers, which was to echo back whatever the value is that is being sent by Terraform. While the value that is returned from the service might be `false` the real value in Azure would be `true`. As of last July the only valid value for the `zoneRedundant` field is `true`.

**Comment by Lobosque** [comment]
We cannot upgrade our module to 4.0 because we get an error due to the provider trying to change a property )`zone_redundant` from` true` to `false` which is immutable:

```
│ Error: creating/updating Namespace (Subscription: "xyz"
│ Resource Group Name: "env-performance"
│ Namespace Name: "env-performance-v1"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: MergeNamespaceFailedWithImmutablePropertyUpdate: ZoneRedundant is an immutable property
│ 
│   with module.service_bus.azurerm_servicebus_namespace.this,
│   on ../service_bus/main.tf line 67, in resource "azurerm_servicebus_namespace" "this":
│   67: resource "azurerm_servicebus_namespace" "this" {
│ 
```

**Comment by WodansSon** [comment]
@Lobosque, I will have to check with the service team to see if they removed the shim in the service. It should be dropping the `zone_redundant` field on the floor, so this should be a no-op.

**Comment by AndersRunningen** [comment]
@nerddtvg From the message above you then mean that this works for `azurerm_eventhub_namespace` resourses or is it something I have missed? 

This when we are trying to apply our Terraform code with terraform version `1.9.6` ad azurerm `4.6.0` we don't really get the update in the plan, but when trying to apply we get the error mentioned in the case. 

When looking at the namespace in `json view` in Azure we can see that the flag is enabled `"zoneRedundant": true,`, but the provider still tries to disable it. 

Have anyone found any workarounds for this?

**Comment by AndersRunningen** [comment]
I just tested creating a new `servicebus_namespace`with `Premium sku`and the value I see in the portal afterwards is still `"zoneRedundant": false `. 

```json
{
    "id": "/subscriptions/3a5fa70a-eee8-45d7-9c55-xxxxxxxxx/resourceGroups/xxxxxxxxx/providers/Microsoft.ServiceBus/namespaces/xxxxxxxxx",
    "name": "xxxxxxxxxx",
    "type": "Microsoft.ServiceBus/Namespaces",
    "location": "westeurope",
    "tags": {
    },
    "properties": {
        "geoDataReplication": {
            "maxReplicationLagDurationInSeconds": 0,
            "locations": [
                {
                    "locationName": "westeurope",
                    "roleType": "Primary",
                    "replicaState": "Ready"
                }
            ]
        },
        "premiumMessagingPartitions": 1,
        "minimumTlsVersion": "1.2",
        "publicNetworkAccess": "Disabled",
        "disableLocalAuth": true,
        "privateEndpointConnections": [
            {
                "id": "/subscriptions/3a5fa70a-eee8-45d7-9c55-xxxxxx/resourceGroups/xxxxxxxxxxx/providers/Microsoft.ServiceBus/namespaces/xxxxxxxxxx/privateEndpointConnections/590fbeb5-7873-4cce-9976-a1f6a1xxxxxx93b5a7",
                "name": "590fbeb5-7873-4cce-9976-a1f6a193xxxxxxxxb5a7",
                "type": "Microsoft.ServiceBus/Namespaces/PrivateEndpointConnections",
                "location": "westeurope",
                "properties": {
                    "provisioningState": "Succeeded",
                    "privateEndpoint": {
                        "id": "/subscriptions/ef0cce31-ae28-4f68-bb2e-xxxxxxxxx/resourceGroups/hub-private-endpoints-rg/providers/Microsoft.Network/privateEndpoints/xxxxxxxxxxxxxxx-private-endpoint"
                    },
                    "privateLinkServiceConnectionState": {
                        "status": "Approved",
                        "description": "Auto-Approved"
                    },
                    "groupIds": [
                        "namespace"
                    ]
                }
            }
        ],
        "zoneRedundant": false,
        "metricId": "3a5fa70a-eee8-45d7-9c55-xxxxxxxx:xxxxxxxxx",
        "serviceBusEndpoint": "https://xxxxxxxxx.servicebus.windows.net:443/",
        "provisioningState": "Succeeded",
        "status": "Active",
        "createdAt": "2024-10-28T11:38:20.6260654Z",
        "updatedAt": "2024-10-28T11:39:18.3083681Z"
    },
    "sku": {
        "name": "Premium",
        "tier": "Premium",
        "capacity": 1
    }
}
```
Terraform: `1.9.6`
AzureRM: `4.6.0`

**Comment by zestmf** [comment]
Commenting here as #27497 was closed. Something has definitely changed on the Azure side in the last few weeks as I just retried applying the same Terraform configuration, with no changes in either my code or the version of TF (`1.9.6`) or AzureRM (`4.3.0`) vs my earlier post [here](https://github.com/hashicorp/terraform-provider-azurerm/issues/27497#issuecomment-2379624906), and it now applies successfully. The Service Bus resource was modified as expected and there was no sign of the MergeNamespaceFailedWithImmutablePropertyUpdate error I was hitting before. So my problem is resolved, but leaving this info in case it is useful.

**Comment by empowerScott** [comment]
> Commenting here as #27497 was closed. Something has definitely changed on the Azure side in the last few weeks as I just retried applying the same Terraform configuration, with no changes in either my code or the version of TF (`1.9.6`) or AzureRM (`4.3.0`) vs my earlier post [here](https://github.com/hashicorp/terraform-provider-azurerm/issues/27497#issuecomment-2379624906), and it now applies successfully. The Service Bus resource was modified as expected and there was no sign of the MergeNamespaceFailedWithImmutablePropertyUpdate error I was hitting before. So my problem is resolved, but leaving this info in case it is useful.

I'm still getting this error:
```
Error: creating/updating Namespace (Subscription: "" Resource Group Name: "" Namespace Name: "prodsb01"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: MergeNamespaceFailedWithImmutablePropertyUpdate: ZoneRedundant is an immutable property
with module.servicebus.azurerm_servicebus_namespace.servicebus1-secondary[0]
on ../modules/web/servicebus/main.tf line 36, in resource "azurerm_servicebus_namespace" "servicebus1-secondary"
```

**Comment by AndersRunningen** [comment]
@nerddtvg don't know if it is correct to tag you? But do you know if there is any updates here?

**Comment by alexeyklots** [comment]
Hi, @WodansSon we also have the same issue with our Premium Service Bus which has enabled Zone redundancy. Basically we have a similar problem as described here https://github.com/hashicorp/terraform-provider-azurerm/issues/27239#issuecomment-2427269988 

Part of our infrastructure was updated to AzureRM v4.x and part not because of this issue with zone redundancy. Maybe there are some workarounds?

**Comment by WodansSon** [comment]
**UPDATE**: I understand this is an issue for a lot of customers and I am working with the service team to figure out the expected behavior. We have opened an internal Microsoft incident around this issue. I will do my best to expedite the resolution of this issue, thank you for your patience and understanding in this matter.

**Comment by WodansSon** [comment]
**ADDITIONAL**: We have confirmed with the service team that this was a service configuration issue and not an issue with the v3.x or v4.x AzureRM providers. The service team has identified the issue and has updated the configuration in our canary environment where we then attempted to repro the scenarios described above, all passed once the service configuration was updated. We are unable to provide an ETA around rolling these updates to all regions in production since we have very narrow windows to deploy currently due to the holiday season.

**Comment by WodansSon** [comment]
**ADDITIONAL ADDITIONAL:**

I have received a communication from the service team with ETA's on release dates for various regions. They are rolling the updates out across 3 phased (see below) with expected release date ETA's per region. 

**NOTE:** _If you cannot wait until the fix is deployed in all regions and you need to create SB Premium namespaces now. You can do so by using an older Terraform AzureRM version(< v4.0 or v3.113 ) and specify `zone_redundant`=`true` while creating the namespace._

## **Phase 1:**
**Release Date:** 11/18/2024

| **Region**        | **Status** |
|:------------------|:-----------|
| BrazilSouth       | Complete   |
| CanadaCentral     | Complete   |
| CentralIndia      | Complete   |
| EastAsia          | Complete   |
| EastUS2EUAP       | Complete   |
| FranceCentral     | Complete   |
| IsraelCentral     | Complete   |
| ItalyNorth        | Complete   |
| JapanWest         | Complete   |
| KoreaCentral      | Complete   |
| MexicoCentral     | Complete   |
| NewZealandNorth   | Complete   |
| SouthAfricaNorth  | Complete   |
| SpainCentral      | Complete   |
| SwedenCentral     | Complete   |
| SwitzerlandNorth  | Complete   |
| UAENorth          | Complete   |
| UKSouth           | Complete   |
| WestUS3           | Complete   |

## **Phase 2:**
**Release Date:** 11/22/2024

| **Region**                  | **Status** |
| :----------------------- | :--------- |
| AustaliaEast                | Pending    |
| CentralUS                   | Pending    |
| GermanyWestCentral | Pending    |
| JapanEast                   | Pending    |
| NorthEurope              | Pending    |
| QatarCentral              | Pending    |
| SoutheastAsia            | Pending    |
| WestEurope               | Pending    |

## **Phase 3:**
**Release Date:** 12/16/2024

| **Region**            | **Status** |
| :------------------- |  :--------- |
| EastUS                  | Pending    |
| EastUS2                | Pending    |
| SouthCentralUS    | Pending    |
| WestUS2               | Pending    |

**Comment by tw-sematell** [comment]
I also had to deploy using azurerm 3.x.x in GermanyWestCentral to enable zone_redundancy. :-/

**Comment by AndersRunningen** [comment]
@WodansSon any update on Zone 2 rollout? Is there any specific settings we need for this to work?

**Comment by WodansSon** [comment]
@AndersRunningen / @tw-sematell , Per the service team Phase 1 and Phase 2 have completed and Phase 3 is due to be live in production on 12/16.

**Comment by WodansSon** [comment]
> @WodansSon any update on Zone 2 rollout? Is there any specific settings we need for this to work?

No, nothing should have to be done on your side as this was a mapping issue between the management plane and the data plane on the service side, the fix should be 100% transparent to you.

**Comment by puneetdevadiga** [comment]
@WodansSon does this resolve the issue for eventhub as well?

**Comment by AndersRunningen** [comment]
I wonder if I am not understanding this correctly, but it doesn't seem to be fixed for `azurerm_servicebus_namespace`. When I am trying to apply on one of our servicebus_namespaces created before `azurerm v3.113`, we can see that the flag in azure for `zone_redundancy = true`. When we are trying to update the servicebus namespace with new configurations with `azurerm 4.12.0` we still get the error:
 
 ```
│ Error: creating/updating Namespace (Subscription: "15f6d4c9-dbec-4d90-bf37-xxxxxxxx"
│ Resource Group Name: "xxxxxxxxxx"
│ Namespace Name: "xxxxxxxxxx"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: MergeNamespaceFailedWithImmutablePropertyUpdate: ZoneRedundant is an immutable property
│ 
│   with module.alz_addons.azurerm_servicebus_namespace.servicebus_namespace["xxxxxxxxxx-event-servicebus"],
│   on .terraform/modules/alz_addons/servicebus-namespace.tf line 11, in resource "azurerm_servicebus_namespace" "servicebus_namespace":
│   11: resource "azurerm_servicebus_namespace" "servicebus_namespace" {
 ```

For me it looks like the Terraform provider is trying the send `zone_redundancy = false` and some imutable logic from before is kicking in. Feel free to explain if I am wrong, just want to update my servicebus 🤷🏼

For new servicebus'es created with 4.X.X versions of the provider it is not an issue as they are all created with the `zone_redundancy` flag = `false` as intended from the above descriptions. 

When asking azure support, they are blind to anything around terraform and say that issue is fixed.

**Comment by krisrmgua** [comment]
just created an eventhub namespace in EastUS2 and still does not have zone redundancy enabled.

**Comment by krisrmgua** [comment]
> **ADDITIONAL ADDITIONAL:**
> 
> I have received a communication from the service team with ETA's on release dates for various regions. They are rolling the updates out across 3 phased (see below) with expected release date ETA's per region.
> 
> **NOTE:** _If you cannot wait until the fix is deployed in all regions and you need to create SB Premium namespaces now. You can do so by using an older Terraform AzureRM version(< v4.0 or v3.113 ) and specify `zone_redundant`=`true` while creating the namespace._
> ## **Phase 1:**
> 
> **Release Date:** 11/18/2024
> **Region** 	**Status**
> BrazilSouth 	Complete
> CanadaCentral 	Complete
> CentralIndia 	Complete
> EastAsia 	Complete
> EastUS2EUAP 	Complete
> FranceCentral 	Complete
> IsraelCentral 	Complete
> ItalyNorth 	Complete
> JapanWest 	Complete
> KoreaCentral 	Complete
> MexicoCentral 	Complete
> NewZealandNorth 	Complete
> SouthAfricaNorth 	Complete
> SpainCentral 	Complete
> SwedenCentral 	Complete
> SwitzerlandNorth 	Complete
> UAENorth 	Complete
> UKSouth 	Complete
> WestUS3 	Complete
> ## **Phase 2:**
> 
> **Release Date:** 11/22/2024
> **Region** 	**Status**
> AustaliaEast 	Complete
> CentralUS 	Complete
> GermanyWestCentral 	Complete
> JapanEast 	Complete
> NorthEurope 	Complete
> QatarCentral 	Complete
> SoutheastAsia 	Complete
> WestEurope 	Complete
> ## **Phase 3:**
> 
> **Release Date:** 12/16/2024
> **Region** 	**Status**
> EastUS 	Pending
> EastUS2 	Pending
> SouthCentralUS 	Pending
> WestUS2 	Pending


just created an Eventhub namespace in EastUS2 and still does not have zone redundancy enabled. The dates for release was Release Date: 12/16/2024. which was yesterday. So not sure why its not working still.

**Comment by WodansSon** [comment]
@krisrmgua, 
> just created an Eventhub namespace in EastUS2 and still does not have zone redundancy enabled. The dates for release was Release Date: 12/16/2024. which was yesterday. So not sure why its not working still. @WodansSon

I will have to reach out to the service team to understand if this has been completed and inform them that our end-users are still hitting this issue when they state that it has been fixed.

**Comment by WodansSon** [comment]
> @WodansSon does this resolve the issue for eventhub as well?

@puneetdevadiga, yes this will resolve both of the issues according to the service team.

**Comment by caridinL6** [comment]
Hi folks, confirming the capacity in `eastus` persists.

One other recommendation to make the documentation in the [provider docs ](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/servicebus_namespace#premium_messaging_partitions-2) regarding the premium message partitions settings. The [documentation here](https://learn.microsoft.com/en-us/rest/api/servicebus/controlplane/namespaces/create-or-update?view=rest-servicebus-controlplane-2024-01-01&tabs=HTTP#sbsku) helped point us in the right direction. To be succinct, the `capacity` value cannot be lower then premium messaging partitions. To take the example from the doc:

If `properties.premiumMessagingPartitions` is 4 then possible `capacity` values are 4, 8, 16, 32 and 64

**Comment by AndersRunningen** [comment]
Can confirm that we were able to run our terraform config for `servicebus-namespace` now. For us this issue seems to be solved as @WodansSon have described over. The change in WestEurope seems to have happened 20.01.2024 from what Azure support mentioned to me outside this issue. 

Thank you for the help 💯

**Comment by tw-sematell** [comment]
Great to hear, thank you everybody!

**Comment by rcskosir** [comment]
:wave: It looks like this has been resolved as noted above. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!

**Comment by BeeDeeBePope** [comment]
Hi @rcskosir, @WodansSon, I'm experiencing the same issue in North Europe region after upgrading azurerm from 3.111.0 to 4.16.0. Is there any confirmation that the fix was deployed there?

**Comment by ranjan-tim** [comment]
Hi @rcskosir, @WodansSon, I'm experiencing the same issue with service bus in China Cloud with AzureRM 4.20 as tested on 24/02. We had to roll back the azureRM to 3.117 for us to unblock.
